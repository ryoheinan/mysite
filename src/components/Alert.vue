<script setup lang="ts">
  interface Props {
    msg: string
    link?: {
      title: string
      url: string
    }
  }
  const props = defineProps<Props>()
</script>

<template>
  <div class="flash">
    {{ props.msg }}
    <a
      v-if="props.link"
      class="btn btn-primary btn-sm flash-action"
      :href="props.link.url"
      target="_blank"
      rel="noopener noreferrer"
      >{{ props.link.title }}</a
    >
  </div>
</template>

<style lang="scss" scoped>
  @import '@primer/css/alerts/index.scss';
</style>
