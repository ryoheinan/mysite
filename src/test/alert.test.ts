import { mount } from '@vue/test-utils'
import Alert from '../components/Alert.vue'
import { describe, expect, it } from 'vitest'

describe('Alert component', () => {
  expect(Alert).toBeTruthy()

  const wrapper = mount(Alert, {
    props: {
      msg: 'こちらからアクセスできます。',
      link: {
        title: 'リンク',
        url: 'https://example.com',
      },
    },
  })

  it('msgが正しく設定されているか', () => {
    expect(wrapper.text()).toContain('こちらからアクセスできます。')
  })
  it('aタグが正しく設定されているか', () => {
    expect(wrapper.find('a').exists()).toBeTruthy()
    expect(wrapper.find('a').text()).toBe('リンク')
    expect(wrapper.find('a').attributes().href).toBe('https://example.com')
  })
  it('Snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
