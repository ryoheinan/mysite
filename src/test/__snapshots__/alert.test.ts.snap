// Vitest Snapshot v1

exports[`Alert component > Snapshot 1`] = `"<div class=\\"flash\\" data-v-d0c93148=\\"\\">こちらからアクセスできます。 <a class=\\"btn btn-primary btn-sm flash-action\\" href=\\"https://example.com\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\" data-v-d0c93148=\\"\\">リンク</a></div>"`;
